### ISSUE TITLE
Add additional hand-written docs

### ISSUE THREAD

**Issue by samhh** [unsplash/sum-types#1]
Including:

- [x] Usage overview for anyone already familiar with the basic idea of sum types
- [ ] Guide for everyone else
- [ ] Deep dive on why we don't support generics
- [ ] How to rename sum type members without missing anything
- [x] Interop with fp-ts `pipe`

**Comment by ericblade** [comment]
Agree. Someone pointed out this library to me with a not very well defined purpose as to why I should look at it, and it's not clear at all what purpose it actually fulfills.

**Comment by samhh** [comment]
@ericblade Does #46 clear things up at all?
